src/types/app-arc56.ts

```diff
@@ -0,0 +1,442 @@
+import algosdk, { ABIValue } from 'algosdk'
+import { ABIReturn } from './app'
+import { Expand } from './expand'
+
+/**
+ * Wrapper around `algosdk.ABIMethod` that represents an ARC-56 ABI method.
+ */
+export class Arc56Method extends algosdk.ABIMethod {
+  override readonly args: Array<
+    Expand<
+      Omit<Method['args'][number], 'type'> & {
+        type: algosdk.ABIArgumentType
+      }
+    >
+  >
+  override readonly returns: Expand<
+    Omit<Method['returns'], 'type'> & {
+      type: algosdk.ABIReturnType
+    }
+  >
+
+  constructor(public method: Method) {
+    super(method)
+    this.args = method.args.map((arg) => ({
+      ...arg,
+      type: algosdk.abiTypeIsTransaction(arg.type) || algosdk.abiTypeIsReference(arg.type) ? arg.type : algosdk.ABIType.from(arg.type),
+    }))
+    this.returns = {
+      ...this.method.returns,
+      type: this.method.returns.type === 'void' ? 'void' : algosdk.ABIType.from(this.method.returns.type),
+    }
+  }
+
+  override toJSON(): Method {
+    return this.method
+  }
+}
+
+/**
+ * Returns the `ABITupleType` for the given ARC-56 struct definition
+ * @param struct The ARC-56 struct definition
+ * @returns The `ABITupleType`
+ */
+export function getABITupleTypeFromABIStructDefinition(struct: StructFields): algosdk.ABITupleType {
```

Comment:
```suggestion
export function getABITupleTypeFromABIStructDefinition(structFields: StructFields): algosdk.ABITupleType {
```

Reply:
In this case we want you to pass a single struct definition, so while the type is called StructFields struct is the right name for the parameter.